src/components/TextInput/TextInput.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import colors from '../../theme/colors.scss';
+
+import './TextInput.scss';
+
+const prefixCls = 'kai-text-input';
+
+class TextInput extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isFocused: false,
+    };
+    this.handleFocusChange = this.handleFocusChange.bind(this);
+  }
+
+  handleFocusChange(isFocused) {
```

Comment:
Saving the cursor/caret position so that the user can continue where they left off once this element refocuses would be ideal

Reply:
Yeah, that shouldn't be too difficult. 👍